projects/SSTorytime/services/sstorytime/module.nix

```diff
@@ -0,0 +1,197 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    mkIf
+    types
+    ;
+
+  cfg = config.services.sstorytime;
+
+  # only create a local database if we're gonna connect to it locally
+  isDatabaseLocal = cfg.database.createLocally && cfg.database.host == "/run/postgresql";
+in
+{
+  options.services.sstorytime = {
+    enable = mkEnableOption "SSTorytime";
+    package = mkPackageOption pkgs "sstorytime" { };
+
+    port = mkOption {
+      type = types.port;
+      description = "Port for the SSTorytime service.";
+      default = 8080;
+    };
+
+    openFirewall = mkEnableOption "the default ports in the firewall for the SSTorytime server.";
+
+    user = mkOption {
+      type = types.nonEmptyStr;
+      default = "sstorytime";
+      description = "User account under which SSTorytime runs.";
+    };
+
+    group = mkOption {
+      type = types.nonEmptyStr;
+      default = "sstorytime";
+      description = "Group under which SSTorytime runs.";
+    };
+
+    database = {
+      createLocally = mkEnableOption "configure a local PostgreSQL database for SSTorytime.";
```

Comment:
This might not be needed. You can automatically decide base on value in `cfg.database.host`.

For example: `localDb = cfg.database.host == "localhost";`